### PR TITLE
Agendas migrate Carto map

### PR DESCRIPTION
**Pull request by jorgeatgu**
Closes https://github.com/PopulateTools/issues/issues/1125

## :v: What does this PR do?

Migrate Carto map to d3 map.

### After this PR

gencat-map

### CODE REVIEW

File: app/javascripts/gencat/modules/gp_gencat_map_controller.js [hunk 1]
```diff
@@ -25,157 +37,309 @@ window.GobiertoPeople.GencatMapController = (function() {
...
+  const { fromDate } = options
+
+  mapboxgl.accessToken = 'pk.eyJ1IjoiYmltdXgiLCJhIjoiY2swbmozcndlMDBjeDNuczNscTZzaXEwYyJ9.oMM71W-skMU6IN0XUZJzGQ';
+
+  let dots
+  let featureElement
+  let initZoom
+  const buttonCloseTooltip = document.getElementById('tooltip--close')
+  buttonCloseTooltip.addEventListener('click', closeTooltip)
+  const dataGenCatTrips = 'https://gencat.gobify.net/api/v1/data/data.csv?sql=select%20*%20from%20trips'
```
Comment (ferblape) [inline app/javascripts/gencat/modules/gp_gencat_map_controller.js:53]: Make this URL relative, to don't depend on the domain (in production is a different one)

**Review by ferblape — CHANGES_REQUESTED** [review]
Screenshot 2020-11-03 at 06 26 07

- [x] Could you remove all the references to CARTO js and CARTO CSS?

- [x] The map for a department doesn't filter the events for that department, i.e https://gencat.gobify.net/ca/departments/departament-d-ensenyament?page=false&start_date=2018-06-01 See in the old code `${departmentCondition}` to filter by department

- [x] You are not filtering either by date, if I select a date range you are showing all the events, but it should be filtered by date. See ` ${dateRangeConditions}` in the old query

**Comment by ferblape** [comment]
Hmmm, it's broken to me when I filter by department, there's a JS error: https://gencat.gobify.net/departamentos/departament-de-cultura?page=false&start_date=2019-11-03

**Comment by ferblape** [comment]
Could you adapt the tooltip a bit to the different sizes? The new one is quite big if the content is just one or two trips, and in FF you can see the scroll:

Screenshot 2020-11-05 at 08 38 54

The old one looked much compact:

Screenshot 2020-11-05 at 08 38 31

**Comment by jorgeatgu** [comment]
Done @blat!

Screenshot 2020-11-05 at 10 15 07

**Pull request merged**